### PR TITLE
Alex M

### PR DESCRIPTION
**Pull request by AlexDjangoX**

### CODE REVIEW

File: src/cohort.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  removeStudentFromCohort (gitHub) {
+    this.students.forEach((item, i) => {
+      if (item.gitHub === gitHub) {
+        this.students.splice(i, 1)
+      }
+      return this.students
+    })
+  }
+}
```
Comment (Creatorrr1) [inline src/cohort.js]: Loved how you gone about the remove method

File: src/manager.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  addCohort (cohortName) {
+    const cohort = new Cohort(cohortName)
+    this.cohorts.push(cohort)
+  }
+
+  addStudent (firstName, lastName, gitHub, cohortName) {
+    const student = new Student(firstName, lastName, gitHub)
+    this.cohorts.forEach((cohort, i) => {
```
Comment (Creatorrr1) [inline src/manager.js:39]: Amazing alternative for for-loop

File: src/manager.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      if (cohortName === cohort.cohortName) {
+        cohort.students.push(student)
+      }
+    })
+  }
```
Comment (Creatorrr1) [inline src/manager.js]: Nice way of pushing the students by cohortName and an amazing alternative for for-loop.

File: src/student.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Student {
+  constructor (firstName, lastName, gitHub) {
+    this.fullName = `${firstName} ${lastName}`
```
Comment (Creatorrr1) [inline src/student.js:3]: Nice way for adding the full name and last name

File: spec/cohort.spec.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        cohort.addStudentToCohort("Alice", "Jones", "XXX");
+        cohort.addStudentToCohort("Alex", "Jones", "ZZZ");
+
+        cohort.removeStudentFromCohort("XXX");
+
+        const result = cohort.students.length;
+
+        // verify
+        expect(result).toEqual(expected);
+    });
```
Comment (Creatorrr1) [inline spec/cohort.spec.js:81]: Nice way of testing the remove method

File: spec/manager.spec.js [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        cohorts.addCohort('AlphaXYZ')
+        cohorts.addCohort('AlphaABC')
+        cohorts.addCohort('AlphaDEF')
+        cohorts.addCohort('AlphaQVW')
+        cohorts.addCohort('AlphaQVW')
+
+        const result = cohorts.removeCohortByName('AlphaABC_no_such_cohort')
+
+        expect(result).toEqual(expected)
+    })
```
Comment (Dan2024) [inline spec/manager.spec.js]: your tests look really clean and easy to read 👍

File: src/student.js (hunk 4 above)
Comment (Dan2024) [reply src/student.js:3]: same as above, i didn't think of doing that but i like the idea

File: src/manager.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        cohort.addStudentToCohort(firstName, lastName, gitHub, uniqueId)
+        addedCohorts = cohort
+      }
+    })
+    if (addedCohorts === null) {
+      return 'Not found'
+    }
+
+    return addedCohorts
+  }
```
Comment (Dan2024) [inline src/manager.js]: line 35: in my opinion making the function name _addStudentToCohort_ or _addStudentUniqueId_**To**_Cohort_ would be a better name - reading that function initially made me think it was going to add another cohort - especially with the addedCohorts variable in there

line 39: whats the i doing here? i can't see it being used anywhere else in the function

**Pull request closed without merging**